src/qmapshack/gis/search/CSearchLineEdit.cpp

```diff
@@ -0,0 +1,162 @@
+#include "CSearchLineEdit.h"
+
+CSearchExplanationDialog* CSearchLineEdit::explanationDlg = nullptr;
+
+CSearchLineEdit::CSearchLineEdit(QWidget *parent, IGisProject * project)
+    : CSearchLineEdit(parent)
+{
+    connectedProject=project;
+}
+
+CSearchLineEdit::CSearchLineEdit(QWidget *parent)
+    : QLineEdit (parent)
+{
+    actionClearFilter = new QAction(QIcon(":/icons/32x32/Filter.png"), tr("Clear Filter"), this);
+    actionHelp = new QAction(QIcon(":/icons/32x32/CSrcUnknown.png"), tr("Clear Filter"), this);
+    actionSetupFilter = new QAction(QIcon(":/icons/32x32/Apply.png"), tr("Setup Filter"), this);
+    actionError = new QAction(QIcon(":/icons/32x32/Attention.png"), tr("Error parsing search"), this);
+    actionAutoProperty = new QAction(QIcon(":/icons/32x32/Hint.png"), tr("Auto selected property"), this);
+
+    actionNameOnly = new QAction(tr("Name Only"), this);
+    actionNameOnly->setCheckable(true);
+    actionCompleteText = new QAction(tr("Complete Text"), this);
+    actionCompleteText->setCheckable(true);
+    actionCaseSensitive = new QAction(tr("Clear Filter"), this);
```

Comment:
the tr string does not match, in current version is "Case Sensitive" 
```suggestion
    actionCaseSensitive = new QAction(tr("Case Sensitive"), this);